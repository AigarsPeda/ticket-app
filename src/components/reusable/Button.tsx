import React from "react";

interface IButton {
  label: string;
  type: "button" | "submit" | "reset";
  className: string;
  disable: boolean;
  handleClick?: () => void;
}

const Button: React.FC<IButton> = (props) => {
  const { label, type, className, disable = false, handleClick } = props;
  return (
    <>
      <button
        type={type}
        className={className}
        onClick={handleClick}
        disabled={disable}
      >
        {label}
      </button>
    </>
  );
};

export default Button;
