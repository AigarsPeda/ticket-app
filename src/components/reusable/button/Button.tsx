import React from "react";

interface IButton {
  label: string;
  type: "button" | "submit" | "reset";
  className: string;
  disable?: boolean;
  handleClick?: () => void;
}

const Button: React.FC<IButton> = (props) => {
  const { label, type, className, disable, handleClick } = props;
  return (
    <button
      data-testid="button"
      type={type}
      className={className}
      onClick={handleClick}
      disabled={disable}
    >
      {label}
    </button>
  );
};

export default Button;
