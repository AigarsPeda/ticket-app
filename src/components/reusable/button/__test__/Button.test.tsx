import React from "react";
import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/matchers";

import renderer from "react-test-renderer";

import Button from "../Button";

afterEach(cleanup);

describe("button component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Button
        label="Click"
        type="button"
        className="btn btn-danger ml-2"
      ></Button>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders button correctly", () => {
    const { getByTestId } = render(
      <Button
        label="Click"
        type="button"
        className="btn btn-danger ml-2"
      ></Button>
    );
    expect(getByTestId("button").textContent).toBe("Click");
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Button
          label="Click"
          type="button"
          className="btn btn-danger ml-2"
        ></Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
