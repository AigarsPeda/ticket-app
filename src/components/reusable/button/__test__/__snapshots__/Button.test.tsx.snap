// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component matches snapshot 1`] = `
<button
  className="btn btn-danger ml-2"
  data-testid="button"
  type="button"
>
  Click
</button>
`;
